internal/schema/schema.go

```diff
@@ -340,6 +343,52 @@ func (s *Schema) Parse(schemaString string, useStringDescriptions bool) error {
 		}
 	}
 
+	for _, iface := range s.interfaces {
+		iface.Interfaces = make([]*Interface, len(iface.interfaceNames))
+
+		for i, intfName := range iface.interfaceNames {
+			t, ok := s.Types[intfName]
+			if !ok {
+				return errors.Errorf("interface %q not found", intfName)
+			}
+			intf, ok := t.(*Interface)
+			if !ok {
+				return errors.Errorf("type %q is not an interface", intfName)
+			}
+			for _, f := range intf.Fields.Names() {
+				if iface.Fields.Get(f) == nil {
+					return errors.Errorf("interface %q expects field %q but %q does not provide it", intfName, f, iface.Name)
+				}
+			}
+
+			iface.Interfaces[i] = intf
+		}
+	}
+
+	for _, iface := range s.interfaces {
```

Comment:
In order to avoid checking interfaces multiple times, you might define a variable of type `map[string][]string` above this `for` statement that contains all the dependencies for each interface. Then you will check every interface exactly once if it is not in the map already. 
However, if I understand correctly, you will need to sort all the interfaces by `len(interfaceNames)` ascending. This way interfaces that implement multiple other interfaces will make use of that map efficiently. This map will be used only once at runtime and then it will be garbage collected.